src/box/lua/net_box.lua

```diff
@@ -1101,10 +1114,18 @@ index_metatable = function(remote)
         check_param_table(opts, REQUEST_OPTION_TYPES)
         local key_is_nil = (key == nil or
                             (type(key) == 'table' and #key == 0))
-        local iterator, offset, limit = check_select_opts(opts, key_is_nil)
-        return (remote:_request(M_SELECT, opts, self.space._format_cdata,
-                                self._stream_id, self.space.id, self.id,
-                                iterator, offset, limit, key))
+        local iterator, offset, limit, _, after, fetch_pos =
+            check_select_opts(opts, key_is_nil)
+        check_select_features(opts, remote)
+        local res
+        local method = fetch_pos and M_SELECT_FETCH_POS or M_SELECT
+        res = (remote:_request(method, opts, self.space._format_cdata,
+                               self._stream_id, self.space.id, self.id,
+                               iterator, offset, limit, key, after, fetch_pos))
+        if type(res) ~= 'table' or not fetch_pos or opts and opts.is_async then
```

Comment:
Why do you need to check the result type? Why not write

```Lua
if fetch_pos and not (opts and opts.is_async) then
    return unpack(res)
else
    return res
end
```

?

Reply:
AFAIU if `buffer` is used without `fetch_pos` we shouldn't unpack the result. If it's used with `fetch_pos` we should. I still don't understand the purpose of this check. Can you give an example?

Reply:
Discussed f2f. If `skip_header` is set, we don't unpack and return the position. So we need this extra check.